### ISSUE TITLE
Parallelizer: Mis-detected scalar reduction variable

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#485]
Compiling this program:
```
import "regent"
local __demand(__parallel)
task AdvanceMesh( Mesh : region(ispace(int3d),float) )
where
   reads writes(Mesh)
do
   for c in Mesh do
     for i = 0, 2 do
       var big = 0.0
       for j = 0, 2 do
         big += 4
       end
     end
   end
end
```
produces this error message:
```
...lis/sw/legion/language/src/regent/check_parallelizable.t:1020: Errors reported during typechecking.
Example.rg:3: partition driven auto-parallelization failed: task must return the scalar reduction result
task AdvanceMesh( Mesh : region(ispace(int3d),float) )
   ^
```

**Comment by magnatelee** [comment]
Should be fixed by 5d04e469. Could you pull and try again?

**Comment by manopapad** [comment]
Fixed
